src/libstd/io/stdio.rs

```diff
@@ -872,6 +872,95 @@ impl fmt::Debug for StderrLock<'_> {
     }
 }
 
+/// Equivalent to the [`input`] method except that
+/// a prompt is printed before capturing input.
+///
+/// ## Note
+/// This function is not the recommended solution to get user input, it should
+/// only be used for simple actions. If you require more explicit control over
+/// getting user input, see the [`Stdin::read_line`] method.
+///
+/// ## Panics
+/// This function currently panics if it does not receive input,
+/// or when input is empty without newline
+///
+/// # Examples
+///
+/// ```
+/// #![feature(input_prompt)]
+/// use std::io::input_prompt;
+///
+/// fn main() {
+///     let user_input = input_prompt("Please enter some text: ");
+///
+///     println!("You typed: {}", user_input);
+/// }
+/// ```
+#[unstable(
+    feature = "input_prompt",
+    reason = "this function may be replaced with a \
+                     more general mechanism",
+    issue = "none"
+)]
+pub fn input_prompt(prompt: &str) -> String {
+    stdout().write(prompt.as_bytes()).unwrap();
+    input()
+}
+
+/// Constructs a new handle to the standard input of the current
+/// process, locks this handle and reads a line of input,
+/// returning a `String` containing the input.
+///
+/// If you need more explicit control over
+/// locking, see the [`Stdin::lock`] and [`Stdout::lock`] methods.
+///
+/// ## Note
+/// This function is not the recommended solution to get user input, it should
+/// only be used for simple actions. If you require more explicit control over
+/// getting user input, see the [`Stdin::read_line`] method.
+///
+/// ## Panics
+/// This function currently panics if it does not receive input,
+/// or when input is empty without newline
+///
+/// # Examples
+///
+/// ```
+/// #![feature(input)]
+/// use std::io::input;
+///
+/// fn main() {
+///     print!("Please enter some text: ");
+///
+///     let user_input = input();
+///
+///     println!("You typed: {}", user_input);
+/// }
+/// ```
+#[unstable(
+    feature = "input",
+    reason = "this function may be replaced with a \
+                     more general mechanism",
+    issue = "none"
+)]
+pub fn input() -> String {
+    let mut input = String::new();
+    let _ = stdout().flush();
+    let len = stdin().read_line(&mut input).expect("Failed to read from stdin");
```

Comment:
I don't think it should proceed if len is 0. If someone suddenly insert EOF using `ctrl-d` then the program might need to handle it differently, I think panicking would be a better choice here.

Reply:
Added `panic!()` for invalid inputs